Tests/Dbal/Logging/BacktraceLoggerTest.php

```diff
@@ -3,15 +3,21 @@
 namespace Doctrine\Bundle\DoctrineBundle\Tests\Dbal\Logging;
 
 use Doctrine\Bundle\DoctrineBundle\Dbal\Logging\BacktraceLogger;
+use Doctrine\DBAL\Logging\DebugStack;
 use PHPUnit\Framework\TestCase;
 
+use function class_exists;
 use function current;
 
 /** @group legacy */
 class BacktraceLoggerTest extends TestCase
 {
     public function testBacktraceLogged(): void
     {
+        if (! class_exists(DebugStack::class)) {
```

Comment:
Hmm I guesss we should deprecate it

Reply:
Yes.

Reply:
The BacktraceLogger class should have been defined as internal. Anyway, it is not used used anymore (except by this test) when the middleware system is available and the logging middleware is available in symfony/doctrine-bridge as we use the middleware in those cases.

Reply:
I'm fine with removing it in 2.11. My believe is that doing loud BC break is better than silent one. If somebody is using this class, it's better they get compile error rather than software keep working and discovering that it's not doing what they intended after a long time.